src/applications/qt3spectralscan.py

```diff
@@ -0,0 +1,313 @@
+import sys
+import os
+
+import tkinter as tk
+from tkinter import ttk
+from tkinter import messagebox
+import tkinter.filedialog as filedialog
+
+from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg
+from matplotlib.figure import Figure
+import matplotlib
+import matplotlib.pyplot as plt
+
+import numpy as np
+import pickle
+import threading
+import nipiezojenapy
+
+# Add the path to the qt3utils folder to sys.path
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'qt3utils', 'datagenerators')))
```

Comment:
For some reason I could not import the spectrometer class from the princeton.py file due to the location of the file not being recognized. Will need some help fixing this

Reply:
I see. So there are two options. One is to use a relative import (which you could google / learn about). But the better approach here, since this is essentially an "external" application is to import via qt3utils package. You can think of qt3spectralscan.py as a user of the qt3utils Package and import modules from qt3utils in the normal way

```
from qt3utils.datagenerators.princeton import Spectrometer
```

Let me know if this doesn't work! :)

Reply:
Not working for some reason. My program has never accept a qt3 import in the same style yours did, still not sure why and this may cause problems down the line

Reply:
Did you pip install in editor mode?

Reply:
> Did you pip install in editor mode?

Step 4? 
https://github.com/qt3uw/qt3-utils#development

Reply:
I believe we did this step together over a Google Meets call. Is there a way for me to verify this?

Reply:
Yes — you’d have to look inside the lib/site-packages  folder inside your python environment. Maybe ‘pip list’ also tells you. 

But the simplest thing may be to pip uninstall qt3utils, then reinstall with -e

Reply:
Will try this